### DISCUSSION TITLE
Make `ViewStore.binding(get:compactSend)` API public

### DISCUSSION THREAD

**Discussion by ivanlisovyi** [pointfreeco/swift-composable-architecture#3281] [Ideas]
Hey folks, 

I'd like to pitch an idea of making the following [ViewStore.binding](https://github.com/pointfreeco/swift-composable-architecture/blob/b5c2a3d7c85522ed69f6f1ff3565c8f0ebe68723/Sources/ComposableArchitecture/ViewStore.swift#L461) & [StackState.path](https://github.com/pointfreeco/swift-composable-architecture/blob/27adbdc1e691b325851f6756bfe5a8afe104fa2a/Sources/ComposableArchitecture/Observation/NavigationStack%2BObservation.swift#L423) APIs public. There seem to be legit cases (well at least the project I'm working on has such cases) when no action needs to be sent depending on the state (just like it's done in `NavigationStackStore`). 

What do you think? 

Happy to submit an MR with this change but wanted to check with the authors & community first.

**Comment by mbrandonw** [comment]
Hi @ivanlisovyi, since `ViewStore` is deprecated it will be removed in 2.0. We don't think we should be adding more API to a tool that will be removed in the future.

But luckily I don't believe it's even necessary for us to publicize the API. I believe you can create your own version of that tool without any access to the library's internals.
